stages/org.osbuild.sshd.config

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python3
+"""
+Configure the SSH daemon.
+"""
+
+import fileinput
+import sys
+
+import osbuild.api
+
+
+SCHEMA = """
+"additionalProperties": false,
+"required": ["config"],
+"properties": {
+  "config": {
+    "additionalProperties": false,
+    "description": "SSHD config options",
+    "type": "object",
+    "properties": {
+      "password_authentication": {
+          "additionalProperties": false,
+          "description": "Enable or disable password authentication.",
+          "type": "boolean"
+      },
+      "challenge_response_authentication": {
+          "additionalProperties": false,
+          "description": "Enable or disable challenge/response authentication.",
+          "type": "boolean"
+      }
+    }
+  }
+}
+"""
+
+def bool_to_yes_no(b):
+  if b:
+    return "yes"
+  else:
+    return "no"
+
+def main(tree, options):
+    sshd_config = options["config"]
+    password_auth = sshd_config.get("password_authentication")
+    challenge_response_auth = sshd_config.get("challenge_response_authentication")
+    changes = {}
+    if password_auth is not None:
+      changes["passwordauthentication"] = bool_to_yes_no(password_auth)
+    if challenge_response_auth is not None:
+      changes["challengeresponseauthentication"] = bool_to_yes_no(challenge_response_auth)
+
+    with fileinput.input(files=(f"{tree}/etc/ssh/sshd_config"), inplace=True, encoding="utf-8") as f:
+      for line in f:
+        line_list = line.split()
+        if len(line_list) == 2:
+          key, current_value = line_list
+          value = changes.pop(key.lower())
+          if current_value != value:
+            print(f"{key} {value}")
+            continue
+        print(line)
+      for key, value in changes:
+        print(f"{key} {value}")
```

Comment:
Didn't know about this `inplace` magic.  Pretty cool.
One issue though is that I think this will add extra newlines when it does `print(line)` to keep existing lines intact.

Reply:
Fixed (by stripping).